### ISSUE TITLE
OpenAI: TypeError: 'NoneType' object cannot be interpreted as an integer

### ISSUE THREAD

**Issue by maik-parloa** [DataDog/dd-trace-py#9283]
### Summary of problem
Exception in ddtrace impedes LLM streaming response.

### Which version of dd-trace-py are you using?
- `admission.datadoghq.com/python-lib.version: v2.8.4`
- `site-packages-ddtrace-py3.11-manylinux2014`

### Which version of pip are you using?
- `pip 24.0 from /usr/local/lib/python3.11/site-packages/pip (python 3.11)`

### Which libraries and their versions are you using?

<details>
  <summary>pip freeze</summary>

  ```
aiohttp==3.9.5
ddsketch==3.0.1
ddtrace==2.8.4
httpx==0.27.0
livekit==0.11.1
livekit-agents==0.6.0
livekit-api==0.5.0
livekit-plugins-openai==0.4.0
openai==1.30.1
pydantic==2.7.1
  ```

</details>

### How can we reproduce your problem?
Use the Voice-Assistant example implementation provided by Livekit: https://github.com/livekit/agents/tree/main/examples/voice-assistant

### What is the result that you get?
- https://github.com/DataDog/dd-trace-py/blob/main/ddtrace/contrib/openai/_endpoint_hooks.py#L143

```
Traceback (most recent call last):
  File \"/home/appuser/.local/lib/python3.11/site-packages/livekit/agents/voice_assistant/assistant.py\", line 754, in _synthesize_task
    async for chunk in data.source:
  File \"/home/appuser/.local/lib/python3.11/site-packages/livekit/plugins/openai/llm.py\", line 73, in __anext__
    async for chunk in self._oai_stream:
  File \"/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.11-manylinux2014/ddtrace/contrib/openai/_endpoint_hooks.py\", line 143, in traced_streamed_response
    streamed_chunks = [[] for _ in range(n)]
                                   ^^^^^^^^
TypeError: 'NoneType' object cannot be interpreted as an integer
```

### What is the result that you expected?
- Local execution and "non-Datadog" execution runs smooth (Kind, Helm)

**Comment by maik-parloa** [comment]
Thank you @emmettbutler. It put the following line into our Dockerfile builds to work around the issue for now:

```Dockerfile
RUN sed -i '152a\            if arg.name not in args and arg.default is not inspect.Parameter.empty:\n                args[arg.name] = arg.default' /home/appuser/.local/lib/python3.11/site-packages/livekit/plugins/openai/llm.py
```

**Comment by maik-parloa** [comment]
@Yun-Kim Do you know when the fix will be available? 🙏

**Comment by Yun-Kim** [comment]
Hi @maik-parloa, we're aiming to release 2.9 later next week, but also we're backporting this fix to 2.8 (which we'll also release next week). Hope this helps!

**Comment by maik-parloa** [comment]
That helps a lot. Thank you for the update 🙌
